ios/NumberTileGame/HighScoreArchiver.m

```diff
@@ -0,0 +1,68 @@
+//
+//  HighScoreArchiver.m
+//  NumberTileGame
+//
+//  Created by Michael Prouty on 10/30/19.
+//
+
+#import "HighScoreArchiver.h"
+static NSString *const highScoreArchivePath = @"highScoreArchive";
+
+@implementation HighScoreArchiver
+
++ (NSString*)filePath {
+    NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
+    NSString *documentsDirectoryPath = [paths objectAtIndex:0];
+    NSString *path = [documentsDirectoryPath stringByAppendingPathComponent:highScoreArchivePath];
+    // NSLog(@"File path: %@", path);
+    return path;
+}
+
++ (void)addScore:(NSNumber*)value {
+    NSLog(@"new score added of: %@", value);
+
+    NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+    dateFormatter.dateStyle = NSDateFormatterShortStyle;
+    dateFormatter.timeStyle = NSDateFormatterShortStyle;
+    NSString *dateString = [dateFormatter stringFromDate:[NSDate date]];
+
+    NSDictionary *score = @{
+                                 @"value" : value,
+                                 @"createdAt" : dateString
+                           };
+    NSArray* scores = self.readScores;
+    scores = [scores arrayByAddingObject:score];
+
+
+
+    // This makes me cry
+    NSSortDescriptor *sortDescriptor = [[NSSortDescriptor alloc]
+                                        initWithKey:@"value"
+                                        ascending:NO];
+    scores = [scores sortedArrayUsingDescriptors:@[sortDescriptor]];
+
+
+    if ([scores count] > 3) {
+        scores = [scores subarrayWithRange:NSMakeRange(0, 3)];
+    }
```

Comment:
Where did this limit of 3 come from? I don't see that in the spec.

Reply:
That's how I imagined it when I wrote the spec, but I guess I didn't make it explicit.  I'm not particularly wedded to it.